### ISSUE TITLE
Plugin Registration Functions Should Be Methods on `PluginConfig`

### ISSUE THREAD

**Issue by bryanculver** [nautobot/nautobot#1624]
### Proposed Changes

Functions like `register_template_extensions` are only used in the `ready()` method of `PluginConfig`. These should be moved into methods (likely private) on `PluginConfig`.

### Justification

Make packaging of code more succinct and explicitly show when a function is only to be used by a specific Class and was not designed for broad spectrum use.

**Comment by bryanculver** [comment]
For consideration though is that loader function should NOT be overloadable by a subclass of `PluginConfig`.

**Comment by glennmatthews** [comment]
Yeah, I'm inclined against this because these functions aren't meant to be overloadable.
